padl/transforms.py

```diff
@@ -723,20 +766,29 @@ def _pd_itercall(self, args, mode: Mode, loader_kwargs: Optional[dict] = None,
         if use_preprocess:
             loader = self.pd_get_loader(args, preprocess, mode, **loader_kwargs)
         else:
-            loader = args
+            loader = list(zip(args, range(len(args))))
 
         def _gen():
-            for batch in loader:
+            for batch, ix in loader:
                 batch = _move_to_device(batch, self.pd_device)
-
                 output = batch
                 if use_forward:
-                    output = forward.pd_call_transform(batch, mode)
+                    try:
+                        output = forward.pd_call_transform(batch, mode)
+                    except Exception as err:
+                        self._pd_trace_error(self._pd_get_error_idx('forward'),
+                                             [args[i] for i in ix])
```

Comment:
I thought `ix` refers to the batch idx when you are using the Pytorch dataloader. Is it actually a list of indices?

Reply:
Yes, `_ItemGetter` has been changed to retrieve also indices.

Reply:
Does that mean `ix` can be a batch of index numbers?

Reply:
yes, it is a batch of index numbers

Reply:
But in the case for `loader = list(enumerate(args))` the `ix` will be a single number right? This means running `[.. for i in ix]` will throw a `TypeError` because we would be iterating over an `int`.

Reply:
ah yes, I see, then what about writing a class like `_ItemGetter` for this case, but that retrieves a li. st of one item? For example: ([0], batch). However, the problem with storing the args on memory is still there isn't it?

Reply:
Yes the memory issue with the args would remain.

Reply:
yes, the code has been changed. Before when `use_preprocess` was False, loader was just `list(enumerate(args))`, so in the statement
`for ix, batch in loader`, so `ix` in this case would be an integer. Then, the the line
```
self._pd_trace_error(self._pd_get_error_idx('forward'), [args[i] for i in ix])
```
would produce a `TypeError`.

Reply:
The memory issue had to do with previously wrapping `args` with `list`, which could force the retrieval of all items in args if it happened to be an iterator. It's been resolved now by using `loader = _BatchGetter(args)`